File: lib/blog/site_content/blogpost.ex
defmodule Blog.SiteContent.Blogpost do
  use Ecto.Schema
  import Ecto.Changeset
  alias Blog.User

  schema "blogposts" do
    field(:description, :string)
    field(:published, :boolean, default: false)
    field(:slug, :string)
    field(:text, :string)
    field(:title, :string)
    belongs_to(:user, User)

    timestamps()
  end

  @doc false
  def changeset(blogpost, attrs) do
    blogpost
    |> cast(attrs, [:title, :description, :text, :slug, :published, :user_id])
    |> foreign_key_constraint(:user_id)
    |> validate_required([:title, :description, :text, :slug, :published])
  end
end

File: lib/blog_web/web/static/src/app/blog/blogpost.ts
export class Blogpost {
  public id: number;
  public user_id: number;
  public description: string;
  public title: string;
  public text: string;
  public slug: string;
  public published: boolean;

  public constructor(aBlogpost) {
    this.id = aBlogpost.id;
    this.user_id = aBlogpost.user_id;
    this.description = aBlogpost.description;
    this.title = aBlogpost.title;
    this.text = aBlogpost.text;
    this.slug = aBlogpost.slug;
    this.published = aBlogpost.published;
  }
}
